pages/index.tsx

```diff
@@ -63,6 +64,60 @@ const AnomaliesCheckbox = ({appSettings, set_appSettings}: {
 	</label>;
 };
 
+const SelectAllCheckbox = ({appSettings, set_appSettings}: {
+	appSettings: TSettings,
+	set_appSettings: (s: TSettings) => void
+}): ReactElement | null => {
+	useEffect((): void => {
+		const isAllSelected = [
+			appSettings.shouldShowMissingTranslations,
+			appSettings.shouldShowIcons,
+			appSettings.shouldShowPrice,
+			appSettings.shouldShowRetirement,
+			appSettings.shouldShowYearnMetaFile,
+			appSettings.shouldShowLedgerLive,
+			appSettings.shouldShowStrategies,
+			appSettings.shouldShowRisk,
+			appSettings.shouldShowRiskScore,
+			appSettings.shouldShowDescriptions,
+			appSettings.shouldShowAPY,
+			appSettings.shouldShowWantTokenDescription
+		].every(Boolean);
+
+		set_appSettings({...appSettings, shouldShowAllFilters: isAllSelected});
+	}, [appSettings, set_appSettings]);
```

Comment:
The use effect may call itself once appSettings is updated

Maybe best to do:
```suggestion
	useEffect((): void => {
		set_appSettings((prev) => {
			const isAllSelected = [
				prev.shouldShowMissingTranslations,
				prev.shouldShowIcons,
				prev.shouldShowPrice,
				prev.shouldShowRetirement,
				prev.shouldShowYearnMetaFile,
				prev.shouldShowLedgerLive,
				prev.shouldShowStrategies,
				prev.shouldShowRisk,
				prev.shouldShowRiskScore,
				prev.shouldShowDescriptions,
				prev.shouldShowAPY,
				prev.shouldShowWantTokenDescription
			].every(Boolean);
			return ({...prev, shouldShowAllFilters: isAllSelected})
		});
	}, [set_appSettings]);
```

Reply:
+ I think `set_appSettings` is not required in effects!

Reply:
Sounds good, had to type it a bit https://github.com/yearn/ySync/pull/113/commits/89896c38d7e0c50cda724e3e907b5d8f5a71a1ec